csharp/test/Ice/adapterDeactivation/AllTests.cs

```diff
@@ -186,6 +186,37 @@ public class AllTests : global::Test.AllTests
                         adapter.destroy();
                     }
 
+                    // Two non-loopback endpoints
+                    communicator.getProperties().setProperty("PHAdapter.Endpoints", "tcp -h * -p 12345:default -h *");
+
+                    communicator.getProperties().setProperty("PHAdapter.PublishedHost", "");
+                    {
+                        ObjectAdapter adapter = communicator.createObjectAdapter("PHAdapter"); // unset
+                        var publishedEndpoints = adapter.getPublishedEndpoints();
+                        test(publishedEndpoints.Length == 2);
+                        test(getUnderlying(publishedEndpoints[0].getInfo()) is IPEndpointInfo ipEndpointInfo0 &&
+                            ipEndpointInfo0.host.Length > 0 &&
+                            ipEndpointInfo0.port == 12345);
+                        test(getUnderlying(publishedEndpoints[1].getInfo()) is IPEndpointInfo ipEndpointInfo1 &&
+                            ipEndpointInfo1.host.Length > 0 &&
+                            ipEndpointInfo1.port != 12345);
+                        adapter.destroy();
+                    }
+
+                    communicator.getProperties().setProperty("PHAdapter.PublishedHost", "test.zeroc.com");
+                    {
+                        ObjectAdapter adapter = communicator.createObjectAdapter("PHAdapter"); // unset
+                        var publishedEndpoints = adapter.getPublishedEndpoints();
```

Comment:
Why is it legal to get the published endpoints, before activating the adapter? Here the ephemeral port would not be assigned yet, or am I missing something.

Reply:
The ephemeral ports are assigned when you _create_ the object adapter. 

It's necessary for this common pattern:
```csharp
var adapter = communicator.createObjectAdapterWithEndpoints("MyAdapter", "tcp -h *"); // no port
var proxy = adapter.add(new GreeterService(), greeterIdentity); // port needs to be assigned at this point
adapter.activate();
```